src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -479,7 +484,26 @@ public async Task<int> SendDotNetStreamAsync(DotNetStreamReference dotNetStreamR
 
         try
         {
-            return await Renderer.Dispatcher.InvokeAsync(async () => await dotNetStreamReference.Stream.ReadAsync(buffer));
+            return await Renderer.Dispatcher.InvokeAsync(async () =>
+            {
+                if (dotNetStreamReference.Stream.CanRead)
```

Comment:
I would replace this with a `try..catch` as that's more general and will cover essentially any situation where reading the stream fails for whatever reason.